Source/DiabloUI/ui_item.h

```diff
@@ -295,15 +297,25 @@ class UiListItem {
 public:
 	UiListItem(const char *text = "", int value = 0, UiFlags uiFlags = UiFlags::None)
 	    : m_text(text)
+	    , args(nullptr)
+	    , m_value(value)
+	    , uiFlags(uiFlags)
+	{
+	}
+
+	UiListItem(const char *text, std::vector<DrawStringFormatArg> *args, int value = 0, UiFlags uiFlags = UiFlags::None)
+	    : m_text(text)
+	    , args(args)
 	    , m_value(value)
 	    , uiFlags(uiFlags)
 	{
 	}
 
-	~UiListItem() {};
+	~UiListItem();
 
 	// private:
 	const char *m_text;
+	std::vector<DrawStringFormatArg> *args;

```

Comment:
Does this need to be a pointer? From what I can tell it is owned by the `UiListItem`, so perhaps it can just be a `vector` insttead of a pointer to one?

Reply:
I needed a pointer to avoid this problem:

> Currently can't use `DrawStringFormatArg` directly in `ui_item.h`, cause it's defined in `text_render.hpp` and `text_render.hpp` includes `ui_item.h`. Currently resolved with forward declaration.

There are possibilities to resolve this, for example we could move the definition of `DrawStringFormatArg`.

Reply:
Changed. Thanks for the adjustments in #3610 🙂